receiver/statsdreceiver/protocol/statsd_parser.go

```diff
@@ -102,30 +102,25 @@ func (p *StatsDParser) Aggregate(line string) error {
 	case "g":
 		_, ok := p.gauges[parsedMetric.description]
 		if !ok {
-			metricPoint := buildPoint(parsedMetric)
-			p.gauges[parsedMetric.description] = buildMetric(parsedMetric, metricPoint)
+			p.gauges[parsedMetric.description] = buildGaugeMetric(parsedMetric.description.name, parsedMetric.unit, parsedMetric.floatvalue, parsedMetric.labelKeys, parsedMetric.labelValues, timeNowFunc())
 		} else {
 			if parsedMetric.addition {
-				savedValue := p.gauges[parsedMetric.description].GetTimeseries()[0].Points[0].GetDoubleValue()
+				savedValue := p.gauges[parsedMetric.description].Metrics().At(0).DoubleGauge().DataPoints().At(0).Value()
 				parsedMetric.floatvalue = parsedMetric.floatvalue + savedValue
-				metricPoint := buildPoint(parsedMetric)
-				p.gauges[parsedMetric.description] = buildMetric(parsedMetric, metricPoint)
+				p.gauges[parsedMetric.description] = buildGaugeMetric(parsedMetric.description.name, parsedMetric.unit, parsedMetric.floatvalue, parsedMetric.labelKeys, parsedMetric.labelValues, timeNowFunc())
 			} else {
-				metricPoint := buildPoint(parsedMetric)
-				p.gauges[parsedMetric.description] = buildMetric(parsedMetric, metricPoint)
+				p.gauges[parsedMetric.description] = buildGaugeMetric(parsedMetric.description.name, parsedMetric.unit, parsedMetric.floatvalue, parsedMetric.labelKeys, parsedMetric.labelValues, timeNowFunc())
 			}
 		}
 
 	case "c":
 		_, ok := p.counters[parsedMetric.description]
 		if !ok {
-			metricPoint := buildPoint(parsedMetric)
-			p.counters[parsedMetric.description] = buildMetric(parsedMetric, metricPoint)
+			p.counters[parsedMetric.description] = buildCounterMetric(parsedMetric.description.name, parsedMetric.unit, parsedMetric.intvalue, parsedMetric.labelKeys, parsedMetric.labelValues, timeNowFunc())
```

Comment:
nit: would it help readability to pass `parsedMetric` to the build function, as opposed to passing all of its fields like this? (I think it would, unless there's a reason not to.)

Reply:
Make sense, are taking a look.

Reply:
Revised.